Indicators/RelativeDailyVolume.cs

```diff
@@ -0,0 +1,145 @@
+    /// cumulative volume to the cumulative volume for a given 
+    /// time of day, measured as a ratio.
+    /// 
+    /// Current volume from open to current time of day / Average over the past x days from open to current time of day
+    /// </summary>
+    public class RelativeDailyVolume : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly Dictionary<DateTime, SimpleMovingAverage> _relativeData;
+        private readonly Dictionary<DateTime, decimal> _currentData;
+        private int _previousDay;
+        private int _days;
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => _days == WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the RelativeDailyVolume class using the specified period
+        /// </summary>
+        /// <param name="period">The period over which to perform the computation</param>
+        /// /// <param name="resolution">The resolution over which to perform to computation</param>
+        public RelativeDailyVolume(int period = 2, Resolution resolution = Resolution.Daily)
+            : this($"RDV({period})", period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new RelativeDailyVolume indicator with the specified period
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        public RelativeDailyVolume(string name, int period)
+            : base(name)
+        {
+            _relativeData = new Dictionary<DateTime, SimpleMovingAverage>();
+            _currentData = new Dictionary<DateTime, decimal>();
+            WarmUpPeriod = period;
+            _previousDay = -1; /// No calendar day can be -1, thus default is not a calendar day
+            _days = -1; /// Will increment by one after first TradeBar, then will increment by one every new day
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A a value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            if (input.Time.Day != _previousDay)
+            {
+                var cummulativeVolume = 0;
+                foreach (var pair in _currentData)
+                {
+                    /// Arbitrary year, month, day
+                    DateTime timeBar = new DateTime(1, 1, 1, pair.Key.Hour, pair.Key.Minute, pair.Key.Second);
+                    if (!_relativeData.ContainsKey(timeBar))
```

Comment:
Cool!
- Can replace `ContainsKey` & `[]` for single `TryGetValue`
- Believe can replace `DateTime timeBar = new DateTime(1, 1, 1, pair.Key.Hour, pair.Key.Minute, pair.Key.Second);` by directly using `pair.Key.TimeOfDay`